src/techui_builder/generate.py

```diff
@@ -290,21 +292,27 @@ def build_groups(self):
         # Create screen
         self.screen_ = Screen.Screen(self.screen_name)
         # create widget and group objects
-        widgets: list[EmbeddedDisplay | ActionButton] = []
 
         # order is an enumeration of the components, used to list them,
         # and serves as functionality in the math for formatting.
         for component in self.screen_components:
             new_widget = self._create_widget(component=component)
             if new_widget is None:
                 continue
-            widgets.append(new_widget)
+            self.widgets.append(new_widget)
 
-        widgets = self.layout_widgets(widgets)
+        if self.widgets == []:
+            print(
+                f"No available widget for function {self.build_groups.__name__}\
+                      for {self.screen_name}"
+            )
```

Comment:
we don't need this print statement here as it is also printed in `_create_widget()`, so just add a comment or something
```suggestion
        if self.widgets == []:
            # No widget found, so just back out
```